### PR TITLE
Change `ref` Type

### PR DESCRIPTION
**Pull request by mcordova47**
This changes the type of the `ref` props from `String` to a callback (`EffectFn1 el Unit`). The element type is different depending on the tag. I looked in [Web.HTML](https://pursuit.purescript.org/packages/purescript-web-html/3.1.0/docs/Web.HTML) for all of the supported types and fell back on `HTMLElement` in the event that I couldn’t find one.

### CODE REVIEW

File: codegen/consts.js [hunk 1]
```diff
@@ -107,7 +171,6 @@ module.exports.types = {
   multiple: "Boolean",
   muted: "Boolean",
   noValidate: "Boolean",
-  // TODO: Replace `JsCallback`s with better representations:
```
Comment (mcordova47) [inline codegen/consts.js:110]: Looks like this was outdated.

**Review by fsoikin — APPROVED** [review]
Nice! 👍

**Comment by CharlesTaylor-CV** [comment]
@mcordova47 I humbly request you evaluate my upcoming PR before merging this. I still think we should allow both function refs and objects refs for use with `elmish-html`.

Edit: https://github.com/collegevine/app/pull/9758

**Comment by CharlesTaylor-CV** [comment]
@mcordova47 Feel free to proceed. I have been enlightened.

**Review by CharlesTaylor-CV — CHANGES_REQUESTED** [review]
I believe this still needs an entry in the ChangeLog.

**Pull request merged**